twill-core/src/main/java/org/apache/twill/internal/TwillContainerLauncher.java

```diff
@@ -177,7 +188,24 @@ protected void doStartUp() {
 
     @Override
     protected void doShutDown() {
-      // No-op
+      // Wait for sometime for the container to stop
+      // TODO: Use configurable value for stop time
+      int maxWaitSecs = Constants.APPLICATION_MAX_STOP_SECONDS;
+      try {
+        if (shutdownLatch.await(maxWaitSecs, TimeUnit.SECONDS)) {
+          return;
+        }
+      } catch (InterruptedException e) {
+        LOG.error("Got exception while waiting for runnable {}, instance {} to stop", runnable, instanceId);
```

Comment:
I think it's better to ignore the interrupted exception, or at least clear the thread interrupt flag before calling `await`. This is because

1. you are not blocking forever anyway
2. we do want to spend max effort to let the contain finish it's job before it gets killed.